File: algorithms/algorithm-complexity/python/big-O/bubble_sort.py
import random

def bubble_sort(list):
	n = len(list)

	for i in range(n):
		for j in range(0, n - i - 1): # O(n) * O(n) = O(n * n) = O(n**2) grows quadratically
			print(f'step {i} - {j}')
			if list[j] > list[j + 1]:
				list[j], list[j + 1] = list[j + 1], list[j]

	return list

if __name__ == '__main__':

	list_size = int(input('What size is the list? '))

	list = [random.randint(0, 100) for i in range(list_size)]
	print(list)

	order_list = bubble_sort(list)
	print(order_list)

File: algorithms/algorithm-complexity/javaScript/big-O/bubbleSort.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
});


readline.question('What size is the list?: ', (val) => {
	let value = parseInt(val);
	let list = Array(value).fill().map(() => Math.round(Math.random() * value))

	console.log(list);

	function bubbleSort(list) {
		for (let i = 0; i < list.length; i++) { 
			for (let j = 0; j < list.length - i - 1; j++) { // O(n) * O(n) = O(n^2) grows quadratically
				if (list[j] > list[j +1]) {
					let temp = list[j];
					list[j] = list[j + 1];
					list[j + 1] = temp;
				}
			}
		}

		return list;
	}

	console.log(bubbleSort(list));

  readline.close();
});
